src/test/java/io/confluent/kafkarest/integration/ClusterTestHarness.java

```diff
@@ -51,19 +57,45 @@
 
   public static final int DEFAULT_NUM_BROKERS = 1;
 
-  // Shared config
-  protected Queue<Integer> ports;
+  /**
+   * Choose a number of random available ports
+   */
+  public static int[] choosePorts(int count) {
+    try {
+      ServerSocket[] sockets = new ServerSocket[count];
+      int[] ports = new int[count];
+      for (int i = 0; i < count; i++) {
+        sockets[i] = new ServerSocket(0);
+        ports[i] = sockets[i].getLocalPort();
+      }
+      for (int i = 0; i < count; i++)
+        sockets[i].close();
+      return ports;
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * Choose an available port
+   */
+  public static int choosePort() {
```

Comment:
Yes, there is some danger. In particular, we're more likely to run into issues if tests are run in parallel. We changed this in Kafka, but it's harder to fix in these because of how the ports are used -- last I looked into this, there wasn't a convenient way to use a random port and extract the bound port. However, at the moment since we're not using a parallel test runner, this approach works ok in practice.